src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java

```diff
@@ -142,14 +142,15 @@ public boolean isReady(@Nonnull JCloudsSlave slave) {
         },
         JNLP {
             @Override
-            public ComputerLauncher createLauncher(@Nonnull JCloudsSlave slave) {
+            public ComputerLauncher createLauncher(@Nonnull JCloudsSlave slave) throws IOException {
+                Jenkins.getActiveInstance().addNode(slave);
```

Comment:
This anti-pattern in Jenkins cloud plugins was messing with things a lot and it took quite a while to get rid of it. Does it really help in your case?